### PR TITLE
runtime-rs: fix set share sandbox pid namespace

### PR DESCRIPTION
**Pull request by Tim-0731-Hzt**
Set the share sandbox pid namepsace from spec

Fixes:#4680
Signed-off-by: Zhongtao Hu <zhongtaohu.tim@linux.alibaba.com>

### CODE REVIEW

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs [hunk 1]
```diff
@@ -400,3 +403,17 @@ fn amend_spec(spec: &mut oci::Spec) -> Result<()> {
 
     Ok(())
 }
+
+fn handle_pid_namespace(namespaces: &[oci::LinuxNamespace]) -> bool {
```
Comment (jodh-intel) [inline src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:407]: Could you add a unit test for this new function?

**Review by jodh-intel — COMMENTED** [review]
Thanks @Tim-0731-Hzt.

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs (hunk 1 above)
Comment (Tim-0731-Hzt) [reply src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:407]: fixed

Comment (bergwolf) [reply src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:407]: I believe the idea of the patch is to copy `handlePidNamespace` from go runtime, in which case additional changes are required:
1. please remove the pidns setting from container spec
2. please keep the code comments of the go runtime as it helps others understand why it is implemented that way

Comment (bergwolf) [reply src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:409]: @Tim-0731-Hzt Thanks for updating. To be specific, I think the following line of comment is necessary:
```
                // host pidns path does not make sense in kata. Let's just align it with
                // sandbox namespace whenever it is set.
```

Also any explanation why we do not remove the pidns setting from the container spec in runtime-rs?

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs [hunk 2]
```diff
@@ -387,6 +388,9 @@ fn amend_spec(spec: &mut oci::Spec) -> Result<()> {
             resource.network = None;
         }
 
+        *sandbox_pidns = handle_pid_namespace(&linux.namespaces);
```
Comment (jodh-intel) [inline src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:391]: This function needs a unit test, but see https://github.com/kata-containers/kata-containers/pull/4692#issuecomment-1189955702

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs [hunk 3]
```diff
@@ -79,7 +79,8 @@ impl Container {
         // process oci spec
         let mut inner = self.inner.write().await;
         let config = &self.config;
-        amend_spec(&mut spec).context("load spec")?;
+        let mut sandbox_pidns = false;
+        amend_spec(&mut spec, &mut sandbox_pidns).context("load spec")?;
```
Comment (quanweiZhou) [inline src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:83]: Suggest to return the `sandbox_pidns` by amend_spec not pass a `mut sandbox_pidns`.

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs [hunk 4]
```diff
@@ -400,3 +404,54 @@ fn amend_spec(spec: &mut oci::Spec) -> Result<()> {
...
+                    return true;
+                }
+            }
+            _ => continue,
+        }
+    }
+    false
+}
+
+#[test]
```
Comment (quanweiZhou) [inline src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:424]: Suggest adding tests in mod
```
#[cfg(test)]
mod tests {
  #[test]
  fn test_***() {}
}
```

**Review by quanweiZhou — APPROVED** [review]
Thanks @Tim-0731-Hzt, LGTM.

**Review by liubin — APPROVED** [review]

File: src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs (hunk 1 above)
Comment (bergwolf) [reply src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:410]: @Tim-0731-Hzt ping, any comment about this question?

> Also any explanation why we do not remove the pidns setting from the container spec in runtime-rs?

Comment (bergwolf) [reply src/runtime-rs/crates/runtimes/virt_container/src/container_manager/container.rs:413]: Ah, I see that it is handled above in line 395. Thanks @Tim-0731-Hzt !

**Review by bergwolf — APPROVED** [review]

**Pull request merged**